### DISCUSSION TITLE
How to get all cloudflare ip addresses or how to prevent "Received X-Forwarded-For header from an untrusted proxy"?

### DISCUSSION THREAD

**Discussion by robs23** [homeassistant-apps/app-cloudflared#255] [Q&A] [answered]
Hi,
Let me start by thanking you for all the good work on the add-on! It's really great option for secure remote access to your local HA instance..
I've been using cloudflared for about a year now and every now and then I'm getting `Received X-Forwarded-For header from an untrusted proxy` because suddenly cloudflare decided to connect to my HA instance from another ip address.. I've been keeping all previous addresses as `trusted_proxies`, but when cloudflare connects from the address I haven't yet added, I can't connect to my HA remotely.. I've looked into web for a list of all cloudflare ip addresses to add them all at once, but none list is complete (and none ever will be). What can I do in future to protect myself against this kind of situations?

**Comment by brenner-tobias** [comment, marked as answer]
I am glad you like the add-on. Regarding the trusted proxy, we have actually set it up in a different way:
You have to add `172.30.33.0/24` as trusted proxy network (as described [here](https://github.com/brenner-tobias/addon-cloudflared/blob/main/cloudflared/DOCS.md#configurationyaml)). This is the local "docker network" of all the add-ons. Now our add-on will get an IP address within that network range (which again, is a local address). HA will then receive the requests from this local "proxy", which is our Cloudflare daemon in the add-on, creating the tunnel. Therefore, you do not need to care about any Cloudflare IPs whatsoever, because all that HA sees is the local proxy that is running the tunnel.
I hope this makes sense, so kindly let me know, where you have been keeping track of the IPs or if I have misunderstood you.

Best
Tobias

**Reply by brenner-tobias** [reply]
FYI generally speaking (though again I do not think this is relevant), Cloudflare publishes their IP ranges [here](https://www.cloudflare.com/ips/), which does include text-lists for [IPv4](https://www.cloudflare.com/ips-v4) and [IPv6](https://www.cloudflare.com/ips-v6).

**Reply by robs23** [reply]
Thanks! That should do the job

**Reply by brenner-tobias** [reply]
You're welcome @robs23 . Nevertheless, I would love to understand the use-case why you need those IP ranges in the firstplace, so do you mind elaborating?

**Reply by robs23** [reply]
Well, the use-case is exactly as given in the initial message i.e. not being able to connect remotely to HA in case of new Cloudflare's ip address.. One remark though, I didn't have this "docker network" proxy in my configuration, I must have overlooked it somehow. I've added it now and, as far as I understood, I should have no further issues
